### PR TITLE
web: remove codemirror from the initial chunk

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

- Part of https://github.com/sourcegraph/sourcegraph/issues/37845
- The culprit of having `codemirror` in the initial chunk is the `useCodeIntel` hook that we inject in `client/web/src/enterprise/EnterpriseWebApp.tsx`. See the screenshot from Statoscope:
    Screenshot 2023-03-13 at 20 52 26
- This PR removes the injected prop from the root of the web application and instead injects it on the route level by introducing a new component `EnterpriseRepositoryFileTreePage`. It's lazy-loaded and respects the boundaries between the OSS and enterprise versions of the app.

## Test plan

1. `cd client/web && pnpm analyze-bundle` to review the Statoscope report.
2. CI bundlesize check
3. manually tested code-intel functionality on this branch.

## App preview:

- [Web](https://sg-web-vb-remove-codemirror-from-initial.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -7.71% (-238.68 kb) 🔽 | -0.30% (-43.74 kb) 🔽 | 1.72% (+194.94 kb) 🔺 | 0.81% (+6) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vb/remove-codemirror-from-initial-chunk/compare-report.html;tab=live_object) for a full comparison between the commits beea8ca and 91baccd or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by philipp-spiess — APPROVED** [review]
Nice! Yes please

File: client/web/src/enterprise/repo/EnterpriseRepositoryFileTreePage.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { FC } from 'react'
+
+import { RepositoryFileTreePageProps, RepositoryFileTreePage } from '../../repo/RepositoryFileTreePage'
+import { useCodeIntel } from '../codeintel/useCodeIntel'
+
+export const EnterpriseRepositoryFileTreePage: FC<RepositoryFileTreePageProps> = props => {
+    return <RepositoryFileTreePage {...props} useCodeIntel={useCodeIntel} />
```
Comment (vovakulikov) [inline client/web/src/enterprise/repo/EnterpriseRepositoryFileTreePage.tsx:7]: Does anyone know why we are passing this useCodeIntel as a prop? Can't we use it right in consumer? I haven't found any place where we inject different implementations for this

Comment (valerybugakov) [reply client/web/src/enterprise/repo/EnterpriseRepositoryFileTreePage.tsx:7]: It's defined in the enterprise part of the codebase. We cannot directly use it in the references panel in the OSS part. That's why we injected it at the top of the tree.

Comment (vovakulikov) [reply client/web/src/enterprise/repo/EnterpriseRepositoryFileTreePage.tsx:7]: Didn't notice this, thanks. This is just sad.

Comment (valerybugakov) [reply client/web/src/enterprise/repo/EnterpriseRepositoryFileTreePage.tsx:7]: 🥲

Comment (vovakulikov) [reply client/web/src/enterprise/repo/EnterpriseRepositoryFileTreePage.tsx:7]: I quickly checked useCodeIntel and haven't found anything big there, should be trivial to resolve as soon as we have separate package for core types

**Pull request merged**